### DISCUSSION TITLE
Github Octernship Doubts

### DISCUSSION THREAD

**Discussion by MridulTi** [community/community#49888] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

Are we supposed to complete the exercise/task given in the Readme of the project before the Project dates begin?.. For Eg, a project I applied to will start in April. Then Am I supposed to complete the task that is written in the Readme of my repo assigned before that time?
Or 
Should I work on this task in April only?

Also, this new repo assigned to me is not forkable so am I supposed to push changes to this new repo directly?

**Comment by bkpecho** [comment, marked as answer]
Hello, brother @MridulTi,

Typically, you should complete the exercise or task mentioned in the README before the project begins. This will help you familiarize yourself with the project and its codebase, and ensure that you are ready to start contributing once the project begins. However, if the project start date is a long time away and the task is not too complex, you could also choose to wait and complete it closer to the project start date.

Regarding the new repository that was assigned to you, if it is not forkable, it means that you will not be able to create a copy of the repository on your GitHub account. Instead, you will need to clone the repository to your local machine, make changes to the code locally, and then push the changes directly to the new repository. Make sure to follow the guidelines provided by the project maintainer regarding how to contribute to the repository.

Hope my answer helps. Have a nice day!

**Comment by chmodshubham** [comment]
hi @bkpecho  #Octernship
can u please help me to clear my doubts, 

a separate private repo is assigned to each contributor where they can push their code and each individual is doing the same code at their respective repositories in different ways, then 

Q1. how the final code will be selected, and on what criteria?
Q2. how contributors can interact with each other if all doing code on an individual basis, isn't this made to bring more developers to the open-source community?
Q3. and how do the coders get paid, if only one final repository will be selected? Is that mean others whose repositories are not selected will not be paid? are their contribution and time no mean to the community?

Don't take me wrong, these are the doubts I currently have. it would be great to have your assistance.

Thank You in Advance

**Comment by VishalMyla** [comment]
Can you tell me the time line of Github Octernship.
